modules/ppcp-button/services.php

```diff
@@ -124,6 +125,17 @@
 	'button.request-data'                         => static function ( ContainerInterface $container ): RequestData {
 		return new RequestData();
 	},
+	'button.endpoint.simulate-cart'               => static function ( ContainerInterface $container ): SimulateCartEndpoint {
+		if ( ! \WC()->cart ) {
+			throw new RuntimeException( 'cant initialize endpoint at this moment' );
```

Comment:
Should mention which endpoint.

Or maybe this check should be done when actually executing? If we want to keep the `$cart` parameter then it can be nullable and set to `WC()->cart` if null when executing.